ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/ConsensualState.kt

```
@@ -0,0 +1,37 @@
+package net.corda.v5.ledger.consensual
+
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.ledger.consensual.transaction.ConsensualLedgerTransaction
+
+/**
+ * A consensual state (or just "state") contains opaque data used by a consensual ledger. It can be thought of as a disk
+ * file that the program can use to persist data across transactions. ConsensualState are immutable: once created they are never
+ * updated, instead, any changes must generate a new successor state.
+ * Consensual states cannot be consumed.
+ */
+@CordaSerializable
+interface ConsensualState {
+    /**
+     * @property participants A _participant_ is any party whose consent is needed to make a Consensual State valid and final.
+     *
+     * Participants are the main and only verification points for Consensual state since they do not have contract code.
+     * Every participant has to be involved and approve the transaction
+     * so that they receive the updated state, and don't end up in a situation where they can no longer use a state
+     * they possess.
+     *
+     * The participants list should normally be derived from the contents of the state.
+     */
+    val participants: List<Party>
+
+    /**
+     * This needs to be provided to let the State verify its internal state and other parts of the encapsulating
+     * Transaction.
+     * TODO(make services injectable (crypto, etc... CORE-5995)
+     *
+     * @param ledgerTransaction encapsulating transaction
+     *
+     * @return result of the verification
+     *
+     */
+    fun verify(ledgerTransaction: ConsensualLedgerTransaction): Boolean
```

This feels a bit odd to me, but that may be because I don't have a good enough internal model of the consensual ledger. What happens if you give this verify function a transaction that's unrelated to the state you're trying to verify for example? Do you need to be able to retrieve the correct transaction in order to get a valid result from this? If so, how do you do that?

---

Details are not super clear/final yet.
We agreed before Christian's holiday, that we'll need a verify function which will need access to the Transaction and probably some services.
We have this ticket to carry-on with the details: https://r3-cev.atlassian.net/browse/CORE-5982

---

I have changed to throw instead of returning Boolean and added some docs.